jetty-server/src/main/java/org/eclipse/jetty/server/AcceptRateLimit.java

```diff
@@ -0,0 +1,217 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2018 Mort Bay Consulting Pty. Ltd.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.server;
+
+import java.nio.channels.SelectableChannel;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jetty.io.EndPoint;
+import org.eclipse.jetty.io.SelectorManager;
+import org.eclipse.jetty.util.annotation.ManagedObject;
+import org.eclipse.jetty.util.annotation.Name;
+import org.eclipse.jetty.util.component.AbstractLifeCycle;
+import org.eclipse.jetty.util.component.Container;
+import org.eclipse.jetty.util.log.Log;
+import org.eclipse.jetty.util.log.Logger;
+import org.eclipse.jetty.util.statistic.RateStatistic;
+import org.eclipse.jetty.util.thread.Scheduler;
+
+/**
+ * <p>A Listener that limits the rate at which new connections are accepted</p>
+ * <p>
+ * If the limits are exceeded, accepting is suspended for the time period, so 
+ * incoming connections are held in the operating system accept queue (no syn ack sent),
+ * where they may either timeout or wait for the server to resume accepting.
+ * It can be applied to an entire server or to a specific connector by adding it 
+ * via {@link Container#addBean(Object)}
+ * </p>
+ * <p>
+ * <b>Usage:</b>
+ * </p>
+ * <pre>
+ *   Server server = new Server();
+ *   server.addBean(new AcceptLimit(100,5,TimeUnit.SECONDS,server));
+ *   ...
+ *   server.start();
+ * </pre>
+ * @see SelectorManager.AcceptListener
+ */
+@ManagedObject
+public class AcceptRateLimit extends AbstractLifeCycle implements SelectorManager.AcceptListener, Runnable
```

Comment:
This class should be really called `AcceptRateLimiter`. See also comment on the XML file related to this class.